lumbermill-aws/src/main/java/lumbermill/internal/aws/S3ClientImpl.java

```diff
@@ -113,13 +121,9 @@ public void init() {
                         LOGGER.debug("Deleted s3 file s3://{}/{} successfully", sBucket, sKey);
                     }
                 })
-                .doOnUnsubscribe(() -> {
+                .doAfterTerminate(() -> {
```

Comment:
Well, this is how it works, see this question in mailing list https://groups.google.com/forum/#!topic/rxjava/QG2GiTAH044.

Lets discuss the best way to solve this.